File: packages/lutrachat_backend/lutrachat_backend_rest/lib/src/structure/controller/authentication.dart
import 'package:shelf/shelf.dart';

import '../../model/http/common/token/response.dart';

/// A controller that performs user authentication.
abstract interface class AuthenticationController {
  /// Creates a new account and retrieves an authentication token for the given credentials.
  Future<AuthenticationTokenResponse> register(Request request);

  /// Retrieves an authentication token for the given credentials.
  Future<AuthenticationTokenResponse> login(Request request);
}

File: internal/app/api/model/authentication.go
package model

// AuthenticationRegisterSchema represents body of authentication register request.
type AuthenticationRegisterSchema struct {
	Name     string `json:"name"`     // The user's name.
	Email    string `json:"email"`    // The user's email address.
	Password string `json:"password"` // The user's password.
}

// AuthenticationLoginSchema represents body of authentication login request.
type AuthenticationLoginSchema struct {
	Email    string `json:"email"`    // The user's email address.
	Password string `json:"password"` // The user's password.
}

// AuthenticationTokenResponse contains authentication token.
type AuthenticationTokenResponse struct {
	Token string `json:"token"` // Authentication token itself.
}
